src/main/java/com/atlassian/db/replica/api/context/Reason.java

```diff
@@ -0,0 +1,43 @@
+package com.atlassian.db.replica.api.context;
+
+import java.util.Objects;
+
+public final class Reason {
+    private final String value;
+
+    private Reason(final String value) {
+        this.value = value;
+    }
+
+    public static final Reason API_CALL = new Reason("API_CALL");
+    public static final Reason REPLICA_INCONSISTENT = new Reason("REPLICA_INCONSISTENT");
+    public static final Reason READ_OPERATION = new Reason("READ_OPERATION");
```

Comment:
It seems that this is silently coupled with `QueryContext.runOnMain = false`.
In other words, if we see a `QueryContext(READ_OPERATION, true)` then it's a bug. Right?
If so, let's make this coupling explicit, keep it local, e.g. by adding the boolean here.

Reply:
- [ ] keep it local, e.g. by adding the boolean here.

Reply:
If we do that then we have to split `API_CALL` to one for `runOnMain = true` and another for `false`